helix-core/src/main/java/org/apache/helix/controller/rebalancer/strategy/crushMapping/CardDealingAdjustmentAlgorithmV2.java

```diff
@@ -55,6 +74,27 @@ public CardDealingAdjustmentAlgorithmV2(Topology topology, int replica, Mode mod
     }
   }
 
+  // expose internal fields only for testing purpose
```

Comment:
As I mentioned in another PR, one trick you can use is to create a MockCardDealingAdjustmentAlgorithmV2 that extends this class and that contains these methods. That way you should be able to access all protected fields, and use the mock class for unit testing.

Reply:
+1 on Hunter's suggestion, that will minimize the production code.